gravity.sh

```diff
@@ -1,476 +1,626 @@
+
+# Use "force-reload" when restarting dnsmasq for everything but Wildcards
+dnsRestart="force-reload"
+
+# Source setupVars from install script
+setupVars="${piholeDir}/setupVars.conf"
 if [[ -f "${setupVars}" ]];then
-  . /etc/pihole/setupVars.conf
+  source "${setupVars}"
+
+  # Remove CIDR mask from IPv4/6 addresses
+  IPV4_ADDRESS="${IPV4_ADDRESS%/*}"
+  IPV6_ADDRESS="${IPV6_ADDRESS%/*}"
 else
-  echo -e "  ${COL_LIGHT_RED}Error: /etc/pihole/setupVars.conf missing. Possible installation failure.${COL_NC}
-  Please run 'pihole -r', and choose the 'reconfigure' option to reconfigure."
+  echo -e "  ${COL_LIGHT_RED}Installation Failure: ${setupVars} does not exist! ${COL_NC}
+  Please run 'pihole -r', and choose the 'reconfigure' option to fix."
   exit 1
 fi
 
-# Remove the /* from the end of the IP addresses
-IPV4_ADDRESS=${IPV4_ADDRESS%/*}
-IPV6_ADDRESS=${IPV6_ADDRESS%/*}
-
-# Variables for various stages of downloading and formatting the list
-basename=pihole
-piholeDir=/etc/${basename}
-adList=${piholeDir}/gravity.list
-blackList=${piholeDir}/black.list
-localList=${piholeDir}/local.list
-justDomainsExtension=domains
-matterAndLight=${basename}.0.matterandlight.txt
-supernova=${basename}.1.supernova.txt
-preEventHorizon=list.preEventHorizon
-eventHorizon=${basename}.2.supernova.txt
-accretionDisc=${basename}.3.accretionDisc.txt
-
-skipDownload=false
-
 # Warn users still using pihole.conf that it no longer has any effect
-if [[ -r ${piholeDir}/pihole.conf ]]; then
-echo -e "  ${COL_LIGHT_RED}pihole.conf file no longer supported. Overrides in this file are ignored.${COL_NC}"
+if [[ -r "${piholeDir}/pihole.conf" ]]; then
+  echo -e "  ${COL_LIGHT_RED}Ignoring overrides specified within pihole.conf! ${COL_NC}"
 fi
 
-###########################
-# Collapse - begin formation of pihole
-gravity_collapse() {
-
-  #New Logic:
-  # Does /etc/pihole/adlists.list exist? If so leave it alone
-  #                                      If not, cp /etc/.pihole/adlists.default /etc/pihole/adlists.list
-  # Read from adlists.list
-
-  # The following two blocks will sort out any missing adlists in the /etc/pihole directory, and remove legacy adlists.default
-  if [[ -f "${adListDefault}" ]] && [[ -f "${adListFile}" ]]; then
-    rm "${adListDefault}"
-  fi
-
-  if [ ! -f "${adListFile}" ]; then
-    cp "${adListRepoDefault}" "${adListFile}"
-  fi
-
-  echo -e "  ${INFO} Neutrino emissions detected..."
-  echo ""
-  local str="Pulling source lists into range"
-  echo -ne "  ${INFO} ${str}..."
+# Determine if DNS resolution is available before proceeding with retrieving blocklists
+gravity_DNSLookup() {
+  local plu
 
-  sources=()
-  while IFS= read -r line || [[ -n "$line" ]]; do
-    # Do not read commented out or blank lines
-    if [[ ${line} = \#* ]] || [[ ! ${line} ]]; then
-      echo "" > /dev/null
-    else
-      sources+=(${line})
+  # Determine if pi.hole can be resolved
+  if ! timeout 10 nslookup pi.hole > /dev/null; then
```

Comment:
On a fresh install, this will always fail since `pi.hole` will return `NXDOMAIN`